google/auth/transport/_mtls_helper.py

```diff
@@ -408,28 +408,22 @@ def client_cert_callback():
 
 
 def check_use_client_cert():
-    """Returns the value of the GOOGLE_API_USE_CLIENT_CERTIFICATE variable,
-    or an inferred value('true' or 'false') if unset.
+    """Returns boolean for whether the client certificate should be used for mTLS.
 
-    This value is meant to be interpreted as a "true" or "false" value
-    representing whether the client certificate should be used, but could be any
-    arbitrary string.
-
-    If GOOGLE_API_USE_CLIENT_CERTIFICATE is unset, the value value will be
-    inferred by reading a file pointed at by GOOGLE_API_CERTIFICATE_CONFIG, and
-    verifying it contains a "workload" section. If so, the function will return
-    "true", otherwise "false".
+    If GOOGLE_API_USE_CLIENT_CERTIFICATE is set to true or false, a corresponding
+    bool value will be returned.
```

Comment:
It looks like the behaviour changed since I suggested this wording. Maybe add this:

```
If the value is set to an unexpected string, it will default to False
```